jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/Collections.java

```diff
@@ -1848,7 +1848,7 @@ public boolean equals(Object o) {
 
         SynchronizedCollection(Collection<E> c) {
             this.c = Objects.requireNonNull(c);
-            mutex = this;
+            mutex = new Object();
```

Comment:
LinkedBlockingQueue.java can stay where it is, we only started maintaining a modified vs unmodified split with the newer OpenJDK sources.

I think this would be the best info on building j2objc: http://j2objc.org/docs/Building-J2ObjC.html
A couple extra tips that I should probably add to that doc:
You can often use "make jre_emul_dist" in place of "make dist" if you don't need junit, guava, etc.
If you only need one or two architectures you can set J2OBJC_ARCHS in your environment to avoid building all 8 archs. eg. 'export J2OBJC_ARCHS="macosx iphone64"'